### ISSUE TITLE
StackVertex + EmbeddingLayer produces NPE during backprop

### ISSUE THREAD

**Issue by raymondhs** [deeplearning4j/deeplearning4j#3335]
Hello, I'm trying to modify the example MultiLayerNetworkExternalErrors.java to use a ComputationGraph and a single/shared embedding layer that concatenates 2 inputs. I am using StackVertex to accomplish this, but I encountered NullPointerException when calling backpropGradient. It seems the epsilon in StackVertex is null. Any clue for this (or if I have missed something)? Thanks.

My code gist: [url](https://gist.github.com/raymondhs/c20a837493840f2512b359bf8587b839)

This is the exception I got:

`Exception in thread "main" java.lang.NullPointerException
	at org.deeplearning4j.nn.graph.vertex.impl.StackVertex.doBackward(StackVertex.java:114)
	at org.deeplearning4j.nn.graph.ComputationGraph.calcBackpropGradients(ComputationGraph.java:1258)
	at org.deeplearning4j.nn.graph.ComputationGraph.backpropGradient(ComputationGraph.java:1212)
	at org.deeplearning4j.examples.misc.externalerrors.MultiLayerNetworkExternalErrors.main(MultiLayerNetworkExternalErrors.java:89)`

**Comment by AlexDBlack** [comment]
Looks like this is an edge case for StackVertex

Specifically:
- Embedding layer has to be the first layer in the net, as it's not possible to backpropagate errors through it (i.e., input is an integer index, not a real-valued array; i.e., you can't calculate the derivative of the loss function with respect to an integer index). Consequently, Embedding layer returns null for the epsilons (backprop error) array
- StackVertex assumes that epsilons are always defined

Best workaround I've got for now: Copy + paste StackVertex code to your project as a custom layer. Then: modify it to return null: i.e., in doBackward, add a line near the start ```if(epsilon == null) return null;```

**Comment by raymondhs** [comment]
Thanks for your suggestion. I tried to do this, but this results in NPE in calcBackpropGradients, due to these lines in ComputationGraph.java:

```java
Pair<Gradient, INDArray[]> pair = current.doBackward(truncatedBPTT);
INDArray[] epsilons = pair.getSecond();
```

**Comment by AlexDBlack** [comment]
Fixed here: https://github.com/deeplearning4j/deeplearning4j/pull/3364
